ext/rugged/rugged.h

```diff
@@ -147,6 +147,12 @@ struct rugged_cb_payload
     int exception;
 };
 
+struct rugged_treecount_cb_payload
```

Comment:
This doesn't need to be in the external header because it's only used for the recursive count. Can you move it next to the function? That'll make the implementation easier to follow.

(the other payloads are here because they are used by different files)